packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import ProgressBar from './ProgressBar';
+import { PlayerClient, PlayerClientEvent } from 'teleport/lib/tdp';
+import { PngFrame, ClientScreenSpec } from 'teleport/lib/tdp/codec';
+import cfg from 'teleport/config';
+import { getAccessToken, getHostName } from 'teleport/services/api';
+import TdpClientCanvas from 'teleport/components/TdpClientCanvas';
+
+export const DesktopPlayer = ({
+  sid,
+  clusterId,
+}: {
+  sid: string;
+  clusterId: string;
+}) => {
+  const { playerClient, tdpCliOnPngFrame, tdpCliOnClientScreenSpec } =
+    useDesktopPlayer({
+      sid,
+      clusterId,
+    });
+
+  return (
+    <StyledPlayer>
+      <TdpClientCanvas
+        tdpCli={playerClient}
+        tdpCliOnPngFrame={tdpCliOnPngFrame}
+        tdpCliOnClientScreenSpec={tdpCliOnClientScreenSpec}
+        onContextMenu={() => true}
+        // overflow: 'hidden' is needed to prevent the canvas from outgrowing the container due to some weird css flex idiosyncracy.
+        // See https://gaurav5430.medium.com/css-flex-positioning-gotchas-child-expands-to-more-than-the-width-allowed-by-the-parent-799c37428dd6.
+        style={{ display: 'flex', flexGrow: 1, overflow: 'hidden' }}
+      />
+      <ProgressBarDesktop playerClient={playerClient} />
+    </StyledPlayer>
+  );
+};
+
+export const ProgressBarDesktop = (props: { playerClient: PlayerClient }) => {
+  const { playerClient } = props;
+
+  const [state, setState] = useState({
+    max: 500, // TODO: total time of the recording in ms
+    min: 0, // TODO: the recording always starts at 0 ms
+    current: 0, // TODO: the current time the playback is at
+    time: '-:--', // TODO: the human readable time the playback is at
+    isPlaying: true, // determines whether play or pause symbol is shown
+    move: () => {
+      // TODO
+      return;
+    },
+    toggle: () => {
+      // emits PlayerClientEvent.TOGGLE_PLAY_PAUSE
+      playerClient.togglePlayPause();
+    },
```

Comment:
We could, but then we also need `playerClient` down below for `playerClient.addListener`

Reply:
i'm not sure what you mean

Reply:
Its easiest for me to just take in the entire `playerClient` as a prop, since I need it for multiple uses in here. It would need to be `<ProgressBar playerClient={playerClient} toggle={playerClient.togglePlayPause} />`, but that seems like an unnecessary additional prop.

Reply:
Never mind I'm wrong, I was mixing up `ProgressBarDesktop` and `ProgressBar`.